### PR TITLE
Feature/requests/request by day

### PR DESCRIPTION
**Pull request by LucasGobatto**
Implementacao da funcao responsavel por fazer o request por uma data e regiao especifica. Ela retorna o tipo ClimateDetails[] com dados do clima ao longo do dia passado como parametro.

### CODE REVIEW

**Review by fiali1 — APPROVED** [review]

File: src/pages/ServicesTests.tsx [hunk 1]
```diff
@@ -1,46 +1,40 @@
-import React, { useEffect, useState } from "react";
+import * as React from 'react';
```
Comment (rodrigo98rm) [inline src/pages/ServicesTests.tsx:1]: I would leave the previous import. This way you need to write 'React' in front of every single useEffect and useState, and possibly other hooks that may eventually be used.

File: src/pages/ServicesTests.tsx [hunk 2]
```diff
@@ -1,46 +1,40 @@
...
-  ClimateDetails,
-} from "../services/services.inteface";
+import { FetchResult, WeatherResponseFormat, ClimateDetails, ImageType } from '../services/services.inteface';
 
-import {
-  RequestByName,
-  RequestByLattAndLong,
-  climateCityDetails,
-} from "../services/api.requests";
+import { requestByName, requestByLattAndLong, climateCityDetails } from '../services/api.requests';
```
Comment (rodrigo98rm) [inline src/pages/ServicesTests.tsx:5]: Nice job changing these to start with lower case

File: src/services/api.requests.ts [hunk 3]
```diff
@@ -1,59 +1,55 @@
...
-      return data;
-    } catch (error) {
-      console.log("JSON Error:", error);
-    }
-  } catch (error) {
-    console.log("Error:", error);
-  }
+  return new Promise((resolve, reject) => {
+    makeRequest(APILink).then(resolve).catch(reject);
+  });
```
Comment (rodrigo98rm) [inline src/services/api.requests.ts:19]: There is no need to return a new Promise here. Same to other occurrences below.

```suggestion
  return makeRequest(APILink);
```

File: src/services/api.requests.ts [hunk 4]
```diff
@@ -1,59 +1,55 @@
...
+}
+
+export function getClientLocalization(): Promise<string | void> | void {
+  return new Promise((resolve, reject) => {
+    navigator.geolocation.getCurrentPosition((position) => {
+      resolve(position.coords.latitude + ',' + position.coords.longitude);
+    });
+  });
+} /*    how to get this reject?
```
Comment (rodrigo98rm) [inline src/services/api.requests.ts:54]: Maybe wrap it with a try/catch? Then put the reject inside the catch block

**Review by rodrigo98rm — CHANGES_REQUESTED** [review]

**Pull request closed without merging**